Submitted By: Pierre Labastie <piete DOT labastie AT neuf DOT fr>
Date: 2020-02-13
Initial Package Version: 12.0.2
Upstream Status: Not applicablepplied
Origin: Upstream
Description: Fix building with make-4.3

From af5c725b8109ce83fc04ef0f8bf6aaf0b50c0441 Mon Sep 17 00:00:00 2001
From: Magnus Ihse Bursie <ihse@openjdk.org>
Date: Thu, 30 Jan 2020 11:08:30 +0100
Subject: [PATCH] 8237879: make 4.3 breaks build

Reviewed-by: erikj, tbell
---
 make/common/MakeBase.gmk | 9 +++++----
 1 file changed, 5 insertions(+), 4 deletions(-)

diff --git a/make/common/MakeBase.gmk b/make/common/MakeBase.gmk
index 8de519fc606..221e9455960 100644
--- a/make/common/MakeBase.gmk
+++ b/make/common/MakeBase.gmk
@@ -525,15 +525,16 @@ DependOnVariableFileName = \
 # Param 2 - (optional) name of file to store value in
 DependOnVariableHelper = \
     $(strip \
-        $(eval -include $(call DependOnVariableFileName, $1, $2)) \
+        $(eval $1_filename := $(call DependOnVariableFileName, $1, $2)) \
+        $(if $(wildcard $($1_filename)), $(eval include $($1_filename))) \
         $(if $(call equals, $(strip $($1)), $(strip $($1_old))),,\
-          $(call MakeDir, $(dir $(call DependOnVariableFileName, $1, $2))) \
+          $(call MakeDir, $(dir $($1_filename))) \
           $(if $(findstring $(LOG_LEVEL), trace), \
               $(info NewVariable $1: >$(strip $($1))<) \
               $(info OldVariable $1: >$(strip $($1_old))<)) \
           $(call WriteFile, $1_old:=$(call DoubleDollar,$(call EscapeHash,$($1))), \
-              $(call DependOnVariableFileName, $1, $2))) \
-        $(call DependOnVariableFileName, $1, $2) \
+              $($1_filename))) \
+        $($1_filename) \
     )
 
 # Main macro
